components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -836,6 +836,8 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
             ]);
 
             if (SnapshotContext.is(context)) {
+                // TODO(janx): Remove snapshot access tracking once we're certain that enforcing repository read access doesn't disrupt the snapshot UX.
+                this.trackEvent(ctx, { event: "snapshot_access_request", properties: { snapshot_id: context.snapshotId } }).catch();
```

Comment:
we try to bundle track calls that follow the same context in one call and have differences contained in the properties. in order to follow this here, it would make sense to give the same name to all three track calls (e.g. `snapshot_access`; we try to follow an `object_verb` naming convention from the perspective of the user, feel free to come up with something here that follows it more closely). then, you can add a property such as `type` that contains whether the event is `requested`, `granted`  or `denied` so that we still capture the same level of information.

else, this looks good to me!